src/cmds/net/telnetd/telnetd.c

```diff
@@ -169,16 +168,14 @@ static void *shell_hnd(void* args) {
 		MD(printf("dup2 STDERR_FILENO error: %d\n", errno));
 	}
 
-	ret = shell_run(shell_lookup("tish"));
-	if (ret != 0) {
-		MD(printf("shell_run error: %d\n", ret));
-	}
+	system("tish");
```

Comment:
`system` (by standard) makes `(v)fork+exec+waitpid`. So most of the time shell_hnd thread just waits for another command completion. What if tish started directly from telnetd_client_handler? I suppose it should work and number of threads will reduce